core/templates/pages/exploration-editor-page/settings-tab/settings-tab.component.ts

```diff
@@ -253,8 +282,37 @@ angular.module('oppia').component('settingsTab', {
 
       ctrl.editRole = function(newMemberUsername, newMemberRole) {
         ctrl.closeRolesForm();
-        ExplorationRightsService.saveRoleChanges(
-          newMemberUsername, newMemberRole);
+        if (!ExplorationRightsService.checkUserAlreadyHasRoles(
+          newMemberUsername)) {
+          ExplorationRightsService.saveRoleChanges(
+            newMemberUsername, newMemberRole);
+          return;
+        }
+        let oldRole = ExplorationRightsService.getOldRole(
+          newMemberUsername);
+        reassignRole(newMemberUsername, newMemberRole, oldRole);
+      };
+
+      ctrl.removeRole = function(memberUsername, memberRole) {
+        AlertsService.clearWarnings();
+
+        $uibModal.open({
+          templateUrl: UrlInterpolationService.getDirectiveTemplateUrl(
+            '/pages/exploration-editor-page/settings-tab/templates/' +
+            'remove-role-confirmation-modal.directive.html'),
+          backdrop: true,
+          resolve: {
+            username: () => memberUsername,
+            role: () => memberRole
+          },
+          controller: 'RemoveRoleConfirmationModalController'
+        }).result.then(function() {
+          ExplorationRightsService.removeRoleAsync(
+            memberUsername);
+        }, function() {
+          AlertsService.clearWarnings();
```

Comment:
I think we generally add a "Note to developers: .." comment in the rejection callback for modals.

https://github.com/oppia/oppia/blob/b60866263dda2eb3ec321f188059c1d2cd1d0fa7/core/templates/components/question-directives/questions-list/questions-list.directive.ts#L263-L265

Is there any reason you have added a "clear warnings" statement here?

Reply:
Done